### ISSUE TITLE
Save-Module doesn't save anything on Windows Subsystem for Linux when using Repository parameter

### ISSUE THREAD

**Issue by splatteredbits** [PowerShell/PowerShellGetv2#408]
Steps to reproduce
------------------

```powershell
New-Item -Path 'PSModules' -ItemType 'Directory'
Save-Module -Name 'Pester' -RequiredVersion '4.6.0' -Repository 'PSGallery' -Path './PSModules'
```

If I omit the `Repository` parameter, the module is saved.

Expected behavior
-----------------

I expected Pester (or the module I choose) to be saved into the `PSModules` directory. 

Actual behavior
---------------

The module is missing from `PSModules`/the path I choose.

Environment data
----------------

```powershell
> $PSVersionTable
Name                           Value
----                           -----
PSVersion                      6.1.1
PSEdition                      Core
GitCommitId                    6.1.1
OS                             Linux 4.4.0-17134-Microsoft #471-Microsoft Fri Dec 07 20:04:00 PST 2018
Platform                       Unix
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

```powershell
> Get-Module -ListAvailable PowerShellGet,PackageManagement
ModuleType Version    Name                                ExportedCommands
---------- -------    ----                                ----------------
Script     1.2.4      PackageManagement                   {Find-Package, Find-PackageProvider, Get-Package, Get-PackageProvider...}
Script     2.0.3      PowerShellGet                       {Find-Command, Find-DscResource, Find-Module, Find-RoleCapability...}
```

```powershell
> Get-PackageProvider -ListAvailable
Name                     Version          DynamicOptions
----                     -------          --------------
NuGet                    3.0.0.1          Destination, ExcludeVersion, Scope, SkipDependencies, Headers, FilterOnTag, Contains, AllowPrereleaseVersions, ConfigFile, SkipValidate
PowerShellGet            1.6.7.0          PackageManagementProvider, Type, Scope, AllowClobber, SkipPublisherCheck, InstallUpdate, NoPathUpdate, AllowPrereleaseVersions, Filter, Tag, Includes, DscResource, RoleCapability, Command, Ac...
```

**Comment by SydneyhSmith** [comment]
Thanks for reporting this @splatteredbits we've noticed a recent pattern of unexpected behavior on Windows Subsytem for Linux and will need to do more investigation

**Comment by edyoung** [comment]
I believe this is fixed in PowerShellGet 2.0.4. The check for gallery availability (which is only done if you pass the -Repository param explicitly) was failing inWSL. That check was removed in 2.0.4. Please update and try it out. If you still see the issue please reactivate.
